container/nginx/confs/tls_proxy.conf

```diff
@@ -23,6 +23,41 @@ server {
     include /etc/nginx/conf.d/shared.conf;
 }
 
+server {
+    listen 443 ssl http2;
+
+    ssl_certificate             /usr/src/app/shared/ssl/node.crt;
+    ssl_certificate_key         /usr/src/app/shared/ssl/node.key;
+    ssl_protocols               TLSv1.2 TLSv1.3;
+    ssl_ciphers                 HIGH:!aNULL:!SHA:!RSA; # SHA256/384 and CAMELLIA enabled for compat
+    ssl_prefer_server_ciphers   on;
+    ssl_session_cache           shared:SSL:500m;
+    ssl_session_timeout         365d;
+    ssl_session_tickets         on;
+    ssl_buffer_size             4k;
+    ssl_stapling                on;
+    ssl_stapling_verify         on;
+    ssl_trusted_certificate     /usr/src/app/shared/ssl/node.crt;
+    ssl_early_data              on;
+
+    # endpoint is ipfs/ipns
+    # assumes gateway host is dweb.link. May not be the case in the future.
+    server_name ~^(?<cid>.+)\.(?<endpoint>.+)\.dweb\.link;
```

Comment:
Will this work even if we don't have a wildcard cert for that domain?

Reply:
GW must be ignoring cert validity

Reply:
`server_name`  just checks `Host` header, TLS shouldn't be involved. (I think)